### PR TITLE
NN-3849: (NN-3232) Limit adjudication history to current booking id only

### PR DESCRIPTION
**Pull request by yasinmustafa**
This change relates to a screen on prison staff hub which lists an offenders adjudications, we want to only show adjudications related the latest `bookingId`. I checked in AppInsights over the last 5 months and PSH was the only service that uses the `{offenderNo}/adjudications` endpoint so it's pretty safe just to change the API implementation.
 https://mojdt.slack.com/archives/CK0QLCQ1G/p1624632703313600

Determining the latest booking ID was done using the following condition `AND OB.BOOKING_SEQ = 1`  - I saw this used elsewhere in the prison-api to determine the latest booking

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt [hunk 1]
```diff
@@ -31,17 +31,17 @@ enum class AdjudicationsRepositorySql(val sql: String) {
...
                                 INNER JOIN AGENCY_LOCATIONS AL ON AL.AGY_LOC_ID = AI.AGY_LOC_ID
                                 INNER JOIN AGENCY_INCIDENT_CHARGES AIC ON AIC.AGENCY_INCIDENT_ID = AIP.AGENCY_INCIDENT_ID AND AIC.PARTY_SEQ = AIP.PARTY_SEQ
-                                WHERE AIP.OFFENDER_BOOK_ID IN (
-                                SELECT OFFENDER_BOOK_ID FROM OFFENDER_BOOKINGS OB
-                                        JOIN OFFENDERS OFF ON OFF.OFFENDER_ID = OB.OFFENDER_ID
-                                        WHERE OFF.OFFENDER_ID_DISPLAY = :offenderNo)
+                                INNER JOIN OFFENDER_BOOKINGS OB ON OB.OFFENDER_BOOK_ID = AIP.OFFENDER_BOOK_ID
+                                INNER JOIN OFFENDERS OFF ON OFF.OFFENDER_ID = OB.OFFENDER_ID
+                        WHERE OFF.OFFENDER_ID_DISPLAY = :offenderNo
+                              AND OB.BOOKING_SEQ = 1
```
Comment (yasinmustafa) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt:41]: Used this `AND OB.BOOKING_SEQ = 1` to determine latest booking

Comment (JamieDraperUK) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt:41]: Should we be checking for active bookings too?

Comment (sp-watson) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt:41]: That's convenient - lucky you

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt [hunk 2]
```diff
@@ -53,19 +53,19 @@ enum class AdjudicationsRepositorySql(val sql: String) {
         ,       CASE WHEN AL.ACTIVE_FLAG = 'Y' THEN 1 ELSE 0 END AS ACTIVE
```
Comment (sp-watson) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt:53]: Maybe add `LATEST` to this name?

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt [hunk 3]
```diff
@@ -31,17 +31,17 @@ enum class AdjudicationsRepositorySql(val sql: String) {
         ,       OO.DESCRIPTION      DESCRIPTION
```
Comment (sp-watson) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt:31]: Maybe add `LATEST` to this name?

**Review by sp-watson — APPROVED** [review]

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt (hunk 1 above)
Comment (yasinmustafa) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt:41]: @JamieDraperUK the ticket  only says the latest booking - but will check with John/Dom

Comment (yasinmustafa) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt:41]: @JamieDraperUK checked with Dom - it is the latest booking they want here

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt (hunk 2 above)
Comment (yasinmustafa) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt:53]: Agreed - done

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt (hunk 3 above)
Comment (yasinmustafa) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/sql/AdjudicationsRepositorySql.kt:31]: agreed - done

**Review by sp-watson — APPROVED** [review]

**Pull request merged**